lib/heap_sort.rb

```diff
@@ -1,8 +1,21 @@
-
+require_relative "min_heap"
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(nlogn) - loops n times based on length of list, and then logn adding or removing
+# Space Complexity: O(nlogn)
 def heapsort(list)
```

Comment:
👍 However your space complexity is O(n) because you add n elements into the heap.